docker/install-dynamic-plugins.py

```diff
@@ -0,0 +1,121 @@
+    if not isinstance(plugins, list):
+        raise InstallException(f'{dynamicPluginsFile} content must be a list')
+
+    globalConfig = {
+      'dynamicPlugins': {
+            'rootDirectory': 'dynamic-plugins-root'
+      }
+    }
+
+    # iterate through the list of plugins
+    for plugin in plugins:
+        package = plugin['package']
+        if package.startswith('./'):
+            package = os.path.join(os.getcwd(), package[2:])
+
+        print('\n======= Installing dynamic plugin', package, flush=True)
+
+        print('\t==> Grabbing package archive through `npm pack`', flush=True)
+        completed = subprocess.run(['npm', 'pack', package], capture_output=True, cwd=dynamicPluginsRoot)
+        if completed.returncode != 0:
+            raise InstallException(f'Error while installing plugin { package } with \'npm pack\' : ' + completed.stderr.decode('utf-8'))
+
+        archive = os.path.join(dynamicPluginsRoot, completed.stdout.decode('utf-8').strip())
+        directory = archive.replace('.tgz', '')
+
+        print('\t==> Removing previous plugin directory', directory, flush=True)
+        shutil.rmtree(directory, ignore_errors=True, onerror=None)
+        os.mkdir(directory)
+
+        print('\t==> Extracting package archive', archive, flush=True)
+        file = tarfile.open(archive, 'r:gz') # NOSONAR
+        # extract the archive content but take care of zip bombs
+        for member in file.getmembers():
+            if member.isreg():
+                if not member.name.startswith('package/'):
+                    raise InstallException("NPM package archive archive does not start with 'package/' as it should: " + member.name)
+
+                if member.size > maxEntrySize:
+                    raise InstallException('Zip bomb detected in ' + member.name)
+
+                # Remove the `package/` prefix from the file name
+                member.name = member.name[8:]
+                file.extract(member, path=directory)
+            elif member.isdir():
+                print('\t\tSkipping directory entry', member.name, flush=True)
+            else:
+                raise InstallException('NPM package archive contains a non regular file: ' + member.name)
+
+        file.close()
+
+        print('\t==> Removing package archive', archive, flush=True)
+        os.remove(archive)
+
+        if 'pluginConfig' not in plugin:
+          print('\t==> Successfully installed dynamic plugin', package, flush=True)
+          continue
+
+        # if some plugin configuration is defined, merge it with the global configuration
+
+        print('\t==> Merging plugin-specific configuration', flush=True)
+        config = plugin['pluginConfig']
+        if config is not None and isinstance(config, dict):
+                merge(config, globalConfig)
+
+        print('\t==> Successfully installed dynamic plugin', package, flush=True)
+
+    yaml.safe_dump(globalConfig, open(dynamicPluginsGlobalConfigFile, 'w'))
+
+main()
```

Comment:
From the link you provided, in the [When Should You Avoid the Name-Main Idiom](https://realpython.com/if-name-main-python/#when-should-you-avoid-the-name-main-idiom) header, I read:

> There are also other scenarios when it’s best to avoid the name-main idiom in Python:
> 
> - A pure script: If you write a script that’s _meant_ to be run as a script, then you can put your code execution into the global namespace without nesting it in the name-main idiom. You _can_ use Python as a scripting language because it doesn’t enforce strong object-oriented patterns. You don’t have to stick to design patterns from other languages when you program in Python.
> 

*meant to be run as a script* is **precisely** what this file is.

I agreed to wrap the content into a `main()` function, because it is known to increase performance.
But I don't see why adding this **Name-Main Idiom** in this file, which would decrease readability IMO.

Reply:
> But I don't see why adding this Name-Main Idiom in this file, which would decrease readability IMO.

If others have dealt with python before they would also know about this idiom, even if it's annoying :(. The reason we may want to add the name-main idiom is for testing purposes. If we want to test this file we should add the name-main idiom. If not, I am fine avoiding the idiom.

Reply:
As for testing trough this idiom, in the same section of the link you provided, I read:

> While this works for small files, it’s generally not considered good practice. It’s not advised to mix tests and code in the same file. Instead, write your tests in a separate file. Following this advice generally makes for a more organized code base. This approach also removes any overhead, such as the need to import `unittest` in your main script file.

In our case I'd rather avoid testing code land in the docker image.

Reply:
> Do we want to add a CI test for this script?

Not in this PR., but this could be in a followup PR :-)

Reply:
> If we want to test this file we should add the name-main idiom. If not, I am fine avoiding the idiom.

So is it OK for you to resolve this conversation then ?